### DISCUSSION TITLE
Generate serializers, views and urls for existing models

### DISCUSSION THREAD

**Discussion by elmkarami** [Abdenasser/dr_scaffold#38] [Ideas]
Creating complex models with `dr_scaffold`'s command line is not a trivial task, how about letting the user define the complex model(s) and have a dedicated command to generate the related views.

**Comment by Abdenasser** [comment]
with rails, you have multiple choices: scaffolding means generating everything (as we do now), and there are other generator commands which you can use to only generate a model, view... separately

**Reply by Abdenasser** [reply]
I mean yes we can generate serializers, views and URLs for existing models but in the same time keep the current implementation for lazy people like me 😆

**Reply by elmkarami** [reply]
Yes this should be a separated feature/command.

**Reply by Abdenasser** [reply]
I'm in, what is the workflow? adding a new command or using the same one... adding an option !

**Reply by elmkarami** [reply]
i believe they should have different commands, but i'm open for suggestions.

**Reply by Abdenasser** [reply]
yes a proper command would be better, with a verbose name what about `generate_for Model` or `some_word_for Model` ?

**Reply by Abdenasser** [reply]
`api_for Model` maybe? as this is gonna be a view, urls and serializer.

**Reply by elmkarami** [reply]
`generate_api path_to_model1,path_to_model2,... [--options]`, we can use the model path (importable python module) or the `app_name.ModelName`

**Reply by Abdenasser** [reply]
I'd prefer to have as few inputs as possible so instead of specifying models, we can only specify apps and then generate the view, serializer, and URLs for the models that don't have them yet, if for any reason I don't want them for a some models I can either remove them (which gonna be easy) or maybe have an `--exclude` option something like:

`generate_api_for APPNAME --exclude [Model1, Model2]`

**Reply by Abdenasser** [reply]
I'm actually open to any suggestions, but let's have a few inputs and a command that is easier to remember and write and especially something that is more like the natural English language

**Reply by elmkarami** [reply]
yes i agree, maybe we could also have an `only` option if the app has a lot of models 
`generate_api_for APPNAME --only [Model1, Model2]`

**Reply by Abdenasser** [reply]
that's a brilliant idea

**Reply by Abdenasser** [reply]
@yezz123 I think the docs we've discussed last time would be very useful if we decided to roll out this new generator... I also believe there will be more generators or commands in the future.
